### DISCUSSION TITLE
`space-x-[number]` didn't work with `flex-row-reverse` on the first box.

### DISCUSSION THREAD

**Discussion by HiroKws** [tailwindlabs/tailwindcss#6768] [Help] [answered]
**What version of Tailwind CSS are you using?**
v3.0.7 and CDN version

**What build tool (or framework if it abstracts the build tool) are you using?**

tailwind

**What version of Node.js are you using?**

v16.13.1

**What browser are you using?**

Chrome, FF

**What operating system are you using?**

openSUSE(linux)

**Reproduction URL**

**Describe your issue**

`space-x-[number]` didn't work with `flex-row-reverse` on the first box.

```
<div class="flex flex-row-reverse space-x-3">
  <button class="">1st(on the right)</button>
  <button class="">2nd</button>
</div>

```
Generated CSS:

```
.space-x-3 > :not([hidden]) ~ :not([hidden]) {
  --tw-space-x-reverse: 0;
  margin-right: calc(0.75rem * var(--tw-space-x-reverse));
  margin-left: calc(0.75rem * calc(1 - var(--tw-space-x-reverse)));
}
```

`--tw-space-x-reverse: 0;` looks like fixed value. if this variable is '1', then work well.

**Comment by DistantMe** [comment, marked as answer]
@HiroKws Have you added the `space-x-reverse` class to the parent? This is mentioned in the documentation.
